src/pineko/scale_variations.py

```diff
@@ -6,55 +6,149 @@
 from . import check
 
 
-def construct_ren_sv_grid(grid, max_as, nf):
-    """Generate renormalization scale variation terms for the given grid, according to the max_as.
+def ren_sv_contrib(m, delta, logpart, which_part, nf):
+    """Return the ren_sv contribution relative to the requested log power and perturbative order contribution (which_part).
 
     Parameters
     ----------
-    grid : pineappl.grid.Grid
-        pineappl grid
-    max_as : int
-        max as order
+    m : int
+        first non zero perturbative order
+    delta : int
+        difference between asked order and m
+    logpart : int
+        power of the log asked
+    which_part : int
+        asked perturbative order contribution
     nf : int
         number of active flavors
-    """
-    # First let's check if the ren_sv are already there
-    sv_as, sv_al = check.contains_ren(grid, max_as, max_al=0)
-    if sv_as:
-        rich.print(f"[green]Renormalization scale variations are already in the grid")
-    ### Extract the correct subgrid and call the function to scale it
-
 
-def LO_ren_var(m, k_R, nf):
-    """LO ren scale variations depending on the first non vanishing order m.
+    Returns
+    -------
+    float
+        renormalization scale variation contribution
+    """
+    if delta == 0:
+        return 0.0
+    elif delta == 1:
+        return -m * beta.beta_qcd((2, 0), nf)
+    elif delta == 2:
+        if which_part == 0:
+            if logpart == 1:
+                return -m * beta.beta_qcd((3, 0), nf)
+            else:
+                return 0.5 * m * (m + 1) * (beta.beta_qcd((2, 0), nf) ** 2)
+        else:
+            return -(m + 1) * beta.beta_qcd((2, 0), nf)
+
+
+def compute_scalefactor(m, nec_order, to_construct_order, nf, kR):
+    """Compute the factor of renormalization scale variation.
 
-    k_R is defined as the log(mu_{R}^{2}/ Q^2).
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    nec_order : tuple(int)
+        tuple of the order for which the sv contribution is asked for
+    to_contruct_order : tuple(int)
+        tuple of the sv order to be constructed
+    nf : int
+        number of active flavors
+    kR : float
+        log of the ratio between renormalization scale and Q
 
-    nf is the number of active flavors.
+    Returns
+    -------
+    float
+        full contribution of ren sv
     """
-    return 1.0
-
+    delta = to_construct_order[0] - m
+    logpart = to_construct_order[2]
+    return (kR**logpart) * ren_sv_contrib(m, delta, logpart, nec_order[0] - m, nf)
 
-def NLO_ren_var(m, k_R, nf):
-    """NLO ren scale variations depending on the first non vanishing order m.
 
-    k_R is defined as the log(mu_{R}^{2}/ Q^2).
+def compute_map_of_necessary_orders(m, delta):
```

Comment:
```suggestion
def compute_orders_map(m, delta):
```
`necessary` is just a bit redundant :)
(and in any case, you are in the `scale_variations.py` module, so the scope is already telling you something you don't have to repeat)